packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -146,16 +149,19 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     _setTween();
   }
 
-  void _handleTapDown(PointerDownEvent event) {
-    _animationController.animateTo(1.0, duration: kFadeOutDuration);
+  Future<Null> _handleTapDown(PointerDownEvent event) async {
+    await _lastAnimation;
+    _lastAnimation = _animationController.animateTo(1.0, duration: kFadeOutDuration);
   }
 
-  void _handleTapUp(PointerUpEvent event) {
-    _animationController.animateTo(0.0, duration: kFadeInDuration);
+  Future<Null> _handleTapUp(PointerUpEvent event) async {
+    await _lastAnimation;
+    _lastAnimation = _animationController.animateTo(0.0, duration: kFadeInDuration);
   }
 
-  void _handleTapCancel(PointerCancelEvent event) {
-    _animationController.animateTo(0.0, duration: kFadeInDuration);
+  Future<Null>  _handleTapCancel(PointerCancelEvent event) async {
+    await _lastAnimation;
+    _lastAnimation = _animationController.animateTo(0.0, duration: kFadeInDuration);
   }
 
   @override
```

Comment:
As stated in the PR description: This change is a response to a failing test. We had a test that verified an animation was in progress when a tap down was immediately followed by a tap up.  This used to result in an animation from 0.0 to 0.0, so no animation at all (which this PR explicitly exposes). So either, we already have a test for this and the change her just adopts the implementation to what the test really expects or the test was incorrect and there shouldn't be an animation (looking at OEM widgets it seems like there should be an animation).